src/haddock/modules/scoring/sasascore/__init__.py

```diff
@@ -0,0 +1,122 @@
+"""
+Surface accessibility scoring module.
+
+This module performs a solvent accessibility analysis based on 
+some user-defined residues that should be buried or accessible.
+
+If a supposedly buried (resp. accessible) residue is accessible (resp. buried),
+the score should increase by one. The lower the final score the more consistent
+the model with the user data.
+
+To run this module, the user must provide a dictionary with the buried and/or accessible
+residues. The keys of the dictionary should be the chain identifiers and the values
+should be lists of residue numbers.
+
+Example:
+
+>>> resdic_buried_A: [1, 2, 3]
+>>> resdic_accessible_B: [4, 5, 6]
+"""
+from pathlib import Path
+import os
+
+from haddock.core.defaults import MODULE_DEFAULT_YAML
+from haddock.core.typing import FilePath
+from haddock.modules import get_engine
+from haddock.modules.scoring import ScoringModule
+from haddock.libs.libontology import PDBFile
+from haddock.libs.libutil import parse_ncores
+from haddock.libs.libparallel import get_index_list, Scheduler
+from haddock.modules.scoring.sasascore.sasascore import (
+    AccScore,
+    extract_data_from_accscore_class
+    )
+
+RECIPE_PATH = Path(__file__).resolve().parent
+DEFAULT_CONFIG = Path(RECIPE_PATH, MODULE_DEFAULT_YAML)
+
+
+class HaddockModule(ScoringModule):
+    """HADDOCK3 module to perform accessibility scoring."""
+
+    name = RECIPE_PATH.name
+
+    def __init__(self,
+                 order: int,
+                 path: Path,
+                 initial_params: FilePath = DEFAULT_CONFIG) -> None:
+        super().__init__(order, path, initial_params)
+
+    @classmethod
+    def confirm_installation(cls) -> None:
+        """Confirm module is installed."""
+        return
+
+    def _run(self) -> None:
+        """Execute module."""
+        try:
+            models_to_score = self.previous_io.retrieve_models(
+                individualize=True
+                )
+        except Exception as e:
+            self.finish_with_error(e)
+        if self.params["mode"] == "mpi":
+            input_ncores_slurm = int(os.getenv('SLURM_CPUS_ON_NODE', 1))
+            input_ncores_pbs = int(os.getenv('PBS_CPUS_ON_NODE', 1))
+            input_ncores = max(input_ncores_slurm, input_ncores_pbs)
+        else:
+            input_ncores = self.params["ncores"]
+        ncores = parse_ncores(input_ncores, njobs=len(models_to_score))
```

Comment:
You should not call `parse_ncores` here, this is already handled by the scheduler:

https://github.com/haddocking/haddock3/blob/f9f008f9ccb3c1ecc9bbaa0f1bf54325dce24559/src/haddock/libs/libparallel.py#L159-L166